### ISSUE TITLE
dataset.sort_key not retained after dataset.split()

### ISSUE THREAD

**Issue by kolloldas** [pytorch/text#278]
Hi I was trying out the new `split()` functionality to split a test set into test and validation set. When the `BucketIterator` is being set up from a newly split dataset the sorting fails because `dataset.sort_key` is None. Looking at the `split()`  function I see that a new instance of type `Dataset` is created in which `sort_key` is None by default:
```
def split():
...
if not stratified:
            train_data, test_data, val_data = rationed_split(self.examples, train_ratio,
                                                             test_ratio, val_ratio, rnd)
            return tuple(Dataset(d, self.fields)
                         for d in (train_data, val_data, test_data) if d)
```

I guess one way would be to explicitly copy the `sort_key` into the new instance. A more generic way could be to make a copy of the current instance and replace only the `examples` attribute but I'm not sure if that is really needed.

Thanks!

**Comment by mttk** [comment]
Correct, thanks for spotting this. I'll issue a fix soon.
